qiita_db/base.py

```diff
@@ -136,6 +139,29 @@ def _check_id(self, id_):
             "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE "
             "{0}_id=%s)".format(self._table), (id_, ))[0]
 
+    def _check_portal(self, id_):
+        """Checks that object is accessible in current portal
+
+        Parameters
+        ----------
+        id_ : object
+            The ID to test
+        """
+        self._check_subclass()
```

Comment:
Can you remove this call from here, and from `_check_id` and add it as a first call in the `__init__` function. Right now it is executed twice and since this is an instance method, if it is in the init it will be tested first.